src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,85 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, {
+  ChangeEvent, Dispatch, FormEvent, SetStateAction, useState,
+} from 'react';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
 import './FindMovie.scss';
 
-export const FindMovie: React.FC = () => {
+type Props = {
+  setMovies: Dispatch<SetStateAction<Movie[]>>,
+};
+
+export const FindMovie: React.FC<Props> = ({ setMovies }) => {
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [query, setQuery] = useState('');
+  const [clicked, setClicked] = useState(false);
+  const [error, setError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [sameMovie, setSameMovie] = useState(false);
+
+  const handleFindMovie = async (event: FormEvent<HTMLFormElement>) => {
+    event?.preventDefault();
+    setClicked(true);
+    setIsLoading(true);
+    const data = await getMovie(query);
+
+    if (!('Error' in data)) {
+      setError(false);
+
+      const {
+        Plot, Poster, Title, imdbID,
+      } = data;
+
+      // eslint-disable-next-line
+      const defaultPic = 'https://via.placeholder.com/360x270.png?text=no%20preview';
+
+      setMovie({
+        title: Title,
+        description: Plot,
+        imdbId: imdbID,
+        imgUrl: (Poster === 'N/A' ? defaultPic : Poster),
+        imdbUrl: `https://www.imdb.com/title/${imdbID}`,
+      });
+    } else {
+      setError(true);
+    }
+
+    setIsLoading(false);
+  };
+
+  const handleMovieQuery = (event: ChangeEvent<HTMLInputElement>) => {
+    setError(false);
+    setQuery(event.target.value);
+    setSameMovie(false);
+  };
+
+  const handleAddMovie = () => {
+    if (movie) {
+      setMovies((currentMovies) => {
+        if (!currentMovies.find(mov => mov.imdbId === movie.imdbId)) {
+          return ([
+            ...currentMovies,
+            movie,
+          ]);
+        }
+
+        setSameMovie(true);
+
+        return currentMovies;
+      });
+    }
+
+    setMovie(null);
+    setQuery('');
+    setClicked(false);
+    setError(false);
+  };
+
   return (
     <>
-      <form className="find-movie">
+      <form onSubmit={(event) => handleFindMovie(event)} className="find-movie">
```

Comment:
`onSubmit` is quite a long argument, better rewrite it this way 👇  It would be easier to read.
```suggestion
      <form 
        onSubmit={(event) => handleFindMovie(event)} 
        className="find-movie"
      >
```